test/tests/functional/pbs_cray_reliable_job_startup.py

```diff
@@ -0,0 +1,200 @@
+        that matches file_pattern
+        """
+        input_file = None
+        for item in self.du.listdir(path=file_path, sudo=True):
+            if fnmatch.fnmatch(item, file_pattern):
+                input_file = item
+                break
+        self.assertTrue(input_file is not None)
+        with PBSLogUtils().open_log(input_file, sudo=True) as f:
+            self.assertTrue(search_str in f.read())
+            self.logger.info("Found \"%s\" in %s" % (search_str, input_file))
+
+    @tags('cray', 'smoke')
+    def test_reliable_job_startup_not_supported_on_cray(self):
+        """
+        A job is started with extra nodes. Mom superior will show no sign
+        of tolerating node failure.  Accounting logs won't have 's' record.
+        Input files to prologue and launch hooks will show the
+        tolerate_node_failures=none value.
+        """
+        # instantiate queuejob hook
+        hook_event = "queuejob"
+        hook_name = "qjob"
+        a = {'event': hook_event, 'enabled': 'true'}
+        self.server.create_import_hook(hook_name, a, self.qjob_hook_body)
+
+        # instantiate execjob_prologue hook
+        hook_event = "execjob_prologue"
+        hook_name = "prolo"
+        a = {'event': hook_event, 'enabled': 'true'}
+        self.server.create_import_hook(hook_name, a, self.prolo_hook_body)
+
+        # instantiate execjob_launch hook
+        hook_event = "execjob_launch"
+        hook_name = "launch"
+        a = {'event': hook_event, 'enabled': 'true'}
+        self.server.create_import_hook(hook_name, a, self.launch_hook_body)
+
+        # Submit a job
+        j = Job(TEST_USER, {ATTR_l + '.select': '1:ncpus=3:mem=2gb:vntype=' +
+                            'cray_compute+1:ncpus=3:mem=2gb:vntype=' +
+                            'cray_compute',
+                            ATTR_l + '.place': 'scatter'})
+        start_time = int(time.time())
+        jid = self.server.submit(j)
+        self.server.expect(JOB, {ATTR_state: 'R'}, id=jid)
+
+        # Check for msg in mom superior logs
+        msg = "no nodes released as job does not tolerate node failures"
+        self.server.expect(JOB, 'exec_host', id=jid, op=SET)
+        job_stat = self.server.status(JOB, id=jid)
+        exechost = job_stat[0]['exec_host'].partition('/')[0]
+        mom_superior = self.moms[exechost]
+        found = mom_superior.log_match(
+            msg, starttime=start_time, regexp=True, max_attempts=10)
+        self.assertTrue(found)
+
+        # Check that 's' record is absent since release_nodes() was not called
+        try:
+            self.server.accounting_match(
+                msg=".*%s;%s;.*" % ('s', jid),
+                regexp=True, n=50, max_attempts=3)
+        except PtlLogMatchError:
+            self.logger.info(
+                "There was no 's' record found for job %s, test passes" % jid)
+
+        # On mom superior check the input files to prologue and launch hooks
```

Comment:
can you add why do we need to check the input files for prologue hook?

Reply:
Done.